### PR TITLE
cmsearch-deoverlap

### PR DESCRIPTION
**Pull request by RZ9082**
adding an older version(0.08) of cmsearch-deoverlap for MGnify pipeline port

### CODE REVIEW

File: tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    <command><![CDATA[
+        ln -s '$clan_information' clan_info;
+        ln -s '$cmsearch_matches' cmsearch_matches;
+        LC_ALL=C perl '$__tool_directory__/cmsearch-deoverlap.pl' --clanin clan_info cmsearch_matches;
+        mv *.deoverlapped '$deoverlapped_matches'
+
+    ]]></command>
+    <inputs>
+        <param type="data" name="clan_information" format="txt" label="clan information" help="Each row in the file contains an RNA clan and its corresponding family members"/>
+        <param type="data" name="cmsearch_matches" format="txt" label="cmsearch output"/>
```
Comment (bgruening) [inline tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:15]: cmsearch can have multiple outputs isn't it, do you need here to be more precise?

Comment (bgruening) [reply tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:15]: according to the perl script this second param can conainer multiple cmsearch results. Is that correct?

Comment (RZ9082) [reply tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml]: > according to the perl script this second param can conainer multiple cmsearch results. Is that correct?

Yes, you are right.

for example if you have the files "1.cmsearch.tblout", "2.cmsearch.tblout" and so on...

and the file "files_list" (containing the files names):
1.cmsearch.tblout
2.cmsearch.tblout
.
.
.

you can pass it as the files list in the command:
LC_ALL=C perl cmsearch-deoverlap.pl --clanin ribo.clan -l files_list

and the tool will generate an output for each file.
But I assume in this case the multiple dataset option in Galaxy can be used, right?

File: tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        #end if
+        &&
+        mv *.deoverlapped '$deoverlapped_matches'
+
+    ]]></command>
+    <inputs>
+        <param type="data" name="clan_information" format="txt" label="clan information" help="Each row in the file contains an RNA clan and its corresponding family members"/>
+        <param type="data" name="cmsearch_matches" format="tabular" label="cmsearch tabular format results (--tblout)"/>
+        <param argument="--maxkeep" type="boolean" truevalue="--maxkeep" falsevalue="" label="keep hits that only overlap with other hits that are not kept [default: remove all hits with higher scoring overlap]" />
+        <param name="dirty" type="boolean" truevalue="--dirty" falsevalue="" label="keep intermediate files (sorted tblout files)" help="(--dirty)"/>
```
Comment (bgruening) [inline tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:26]: please use singular here and above

File: tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    </inputs>
+    <outputs>
+        <data name="deoverlapped_matches" format="tabular" label="Deoverlapped matches" />
+        <data name="matches_sort" format="tabular" label="Sorted matches">
+            <filter>dirty</filter>
+        </data>
+    </outputs>
+    <tests>
+        <test>
```
Comment (bgruening) [inline tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:35]: please specify here the expected number of outputs.

File: tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml (hunk 1 above)
Comment (bgruening) [reply tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml]: Ah, so if you out N inputs, it creates N outputs?

Ok, then leave it like that. Nice!

Please change the help text, to singular. So not result__s__

Comment (RZ9082) [reply tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml]: yes, exactly.

File: tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+contig--2255364      -         SSU_rRNA_bacteria    RF00177   hmm      325     1084      765        1      -     -    6 0.52   3.9  624.0  2.5e-185 !   -
+
+contig--2271497      -         SSU_rRNA_bacteria    RF00177   hmm        4      812      835        2      -     -    6 0.51   5.0  597.5  2.5e-177 !   -
+
+contig--2369838      -         SSU_rRNA_bacteria    RF00177   hmm       90      938       20      855      +     -    6 0.52   7.0  609.3  6.6e-181 !   -
+
+
+    ]]></help>
+    <citations>
+        <citation type="doi">10.1093/nar/gkac1080</citation>
```
Comment (bgruening) [inline tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml:90]: Is the citation correct?

Comment (RZ9082) [reply tools/rna_tools/cmsearch_deoverlap/cmsearch-deoverlap.xml]: No, this is the old wrong one. I just updated it.
Thank you for reminding me :)

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks @RZ9082! good work!

**Pull request merged**